Typos and englishify federation

pigmej opened the pull request:
**What this PR does / why we need it**: Typos + "englishify" /federated

**Which issue this PR fixes** None

**Special notes for your reviewer**: Just typos + "englishify" federated and obviously autogenerated stuff.

**Release note**:

<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->

`NONE`

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32278)

<!-- Reviewable:end -->

pigmej committed:
Typos and englishify federation

quinton-hoole commented:
Reviewed 24 of 25 files at r1.
Review status: 24 of 25 files reviewed at latest revision, 2 unresolved discussions.

---

*[federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs/clouddns.go, line 28 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278#-KR9Se8m51YnTvsk4ehR:-KR9Se8m51YnTvsk4ehS:b-1bv6fx) ([raw file](https://github.com/kubernetes/kubernetes/blob/e70a8aacd37bfb12365aff762a5e6d1c7a3ae575/federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs/clouddns.go#L28)):*
> ```Go
> 
> type (
> 	// TODO: We don't' need these yet, so they remain unimplemented.  Add later as required.
> ```

Trailing ' can be removed.

---

*[federation/pkg/federation-controller/service/dns.go, line 138 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278#-KR9TKAMEvEI9YBk09tP:-KR9TKAMEvEI9YBk09tQ:b-ut3vwq) ([raw file](https://github.com/kubernetes/kubernetes/blob/e70a8aacd37bfb12365aff762a5e6d1c7a3ae575/federation/pkg/federation-controller/service/dns.go#L138)):*
> ```Go
> 		if net.ParseIP(endpoint) == nil {
> 			// It's not a valid IP address, so assume it's a DNS name, and try to resolve it,
> 			// replacing it's DNS name with its IP addresses in expandedEndpoints
> ```

Consistent use of its or it's.  I believe that 'its' is the correct form,  wierdly enough :-).

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278)*
<!-- Sent from Reviewable.io -->

quinton-hoole commented:
Reviewed 1 of 25 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278)*
<!-- Sent from Reviewable.io -->

quinton-hoole commented:
Many thanks @pigmej !  A few minor typos to fix, and then this can merge.

pigmej commented:
Review status: 23 of 25 files reviewed at latest revision, 2 unresolved discussions.

---

*[federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs/clouddns.go, line 28 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278#-KR9Se8m51YnTvsk4ehR:-KR9V_D5YET_Zu0Pytx2:bt2vr1b) ([raw file](https://github.com/kubernetes/kubernetes/blob/e70a8aacd37bfb12365aff762a5e6d1c7a3ae575/federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs/clouddns.go#L28)):*
<details><summary><i>Previously, quinton-hoole (Quinton Hoole) wrote…</i></summary>
> Trailing ' can be removed.
</details>
Fixed.

---

*[federation/pkg/federation-controller/service/dns.go, line 138 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278#-KR9TKAMEvEI9YBk09tP:-KR9VbtvlQqUYBJsRUD9:b-w2ocw) ([raw file](https://github.com/kubernetes/kubernetes/blob/e70a8aacd37bfb12365aff762a5e6d1c7a3ae575/federation/pkg/federation-controller/service/dns.go#L138)):*
<details><summary><i>Previously, quinton-hoole (Quinton Hoole) wrote…</i></summary>
> Consistent use of its or it's.  I believe that 'its' is the correct form,  wierdly enough :-).
</details>
Yeah it should be its. Fixed also

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278)*
<!-- Sent from Reviewable.io -->

pigmej commented:
Review status: 23 of 25 files reviewed at latest revision, 2 unresolved discussions.

---

*[federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs/clouddns.go, line 28 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278#-KR9Se8m51YnTvsk4ehR:-KR9Vj7lpG3dLRWKO162:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/e70a8aacd37bfb12365aff762a5e6d1c7a3ae575/federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs/clouddns.go#L28)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary>
> Fixed.
</details>
Done.

---

*[federation/pkg/federation-controller/service/dns.go, line 138 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278#-KR9TKAMEvEI9YBk09tP:-KR9ViFRySj2OSWkMoTK:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/e70a8aacd37bfb12365aff762a5e6d1c7a3ae575/federation/pkg/federation-controller/service/dns.go#L138)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary>
> Yeah it should be its. Fixed also
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278)*
<!-- Sent from Reviewable.io -->

pigmej commented:
@quinton-hoole fixed.

quinton-hoole commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32278)*
<!-- Sent from Reviewable.io -->

nikhiljindal commented on federation/apis/swagger-spec/extensions_v1beta1.json:
```diff
@@ -94,7 +94,7 @@
       "type": "v1beta1.Ingress",
       "method": "POST",
-      "summary": "create a Ingress",
+      "summary": "create an Ingress",
```
swagger spec is autogenerated.
This change will be overwritten when someone runs hack/update-federation-swagger-spec.sh

The right way to fix it is to fix the code that generates it.
Filed https://github.com/kubernetes/kubernetes/issues/32305 with details.
Feel to reply on the issue if you want to take it up.

nikhiljindal commented on federation/docs/api-reference/extensions/v1beta1/definitions.html:
```diff
@@ -951,7 +951,7 @@ <h3 id="_v1_fcvolumesource">v1.FCVolumeSource</h3>
 <tr>
 <td class="tableblock halign-left valign-top"><p class="tableblock">targetWWNs</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">Required: FC target world wide names (WWNs)</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">Required: FC target worldwide names (WWNs)</p></td>
```
Same this is also autogenerated. Will be overriten when someone runs hack/update-federation-api-ref-docs.sh

nikhiljindal commented on federation/docs/api-reference/federation/v1beta1/definitions.html:
```diff
@@ -685,7 +685,7 @@ <h3 id="_v1beta1_clusterstatus">v1beta1.ClusterStatus</h3>
 <tr>
 <td class="tableblock halign-left valign-top"><p class="tableblock">zones</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">Zones is the list of avaliability zones in which the nodes of the cluster exist, e.g. <em>us-east1-a</em>. These will always be in the same region.</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">Zones is the list of availability zones in which the nodes of the cluster exist, e.g. <em>us-east1-a</em>. These will always be in the same region.</p></td>
```
This is coming from `federation/apis/federation/types.go`.
Fix that file to fix this

nikhiljindal commented:
api-ref and swagger spec are autogenerated and hence will be overwriten when we run the script again, so you need to fix the underlying code to fix those files.

Am curious if you found and fixed these manually or you used some tool to do that?

quinton-hoole commented:
LGTM removed.  Thanks @nikhiljindal - I missed those auto-generated files.  @pigmej lets fix the source files and regenerate, as @nikhiljindal suggests.  Then I'll re-add the LGTM label and this should auto-merge.

pigmej commented:
@quinton-hoole @nikhiljindal updated, removed all from `api-reference` and `swagger-spec`.

nikhiljindal commented:
LGTM, thanks!
